lib/ansible/modules/network/net_static_route.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2017, Ansible by Red Hat, inc
+#
+# This file is part of Ansible by Red Hat
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'core'}
+
+
+DOCUMENTATION = """
+---
+module: net_static_route
+version_added: "2.4"
+author: "Ricardo Carrillo Cruz (@rcarrillocruz)"
+short_description: Manage static IP routes on network devices
+description:
+  - This module provides declarative management of static
+    IP routes on network devices.
+options:
+  prefix:
+    description:
+      - Network prefix of the static route.
+    required: true
+  mask:
+    description:
+      - Network prefix mask of the static route.
+    required: true
+  next_hop:
+    description:
+      - Next hop IP of the static route.
+    required: true
+  admin_distance:
+    description:
+      - Admin distance of the static route.
+  collection:
+    description: List of static route definitions
+  purge:
+    description:
+      - Purge static routes not defined in the collections parameter.
+    default: no
+  state:
+    description:
+      - State of the static route configuration.
+    default: present
+    choices: ['present', 'absent']
+"""
+
+EXAMPLES = """
```

Comment:
Could collections examples be added in?

I know https://github.com/ansible/ansible/pull/25210 hasn't been merged yet, though if you know that structure that would have us from having to come back to this